pkg/gui/app_status_manager.go

```diff
@@ -28,6 +34,9 @@ func (m *statusManager) removeStatus(name string) {
 }
 
 func (m *statusManager) addWaitingStatus(name string) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
 	m.removeStatus(name)
```

Comment:
Given that removeStatus tries to obtain the lock that addWaitingStatus already holds, wouldn't that cause a deadlock?